packages/@o3r/localization/schematics/add-localization-key/index.spec.ts

```diff
@@ -0,0 +1,160 @@
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+      const tree = await runner.runSchematic('add-localization-key', {
+        path: o3rComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, initialTree);
+
+      const translationFileContent = tree.readText(translationPath);
+      expect(translationFileContent).toContain('dummyLoc1: string;');
+      expect(translationFileContent).toContain('dummyLoc1: \'o3r-test-pres.dummyLoc1\'');
+
+      const localizationFileContent = tree.readJson(localizationPath);
+      expect(localizationFileContent['o3r-test-pres.dummyLoc1'].description).toBe('Dummy 1 description');
+      expect(localizationFileContent['o3r-test-pres.dummyLoc1'].defaultValue).toBe('Dummy 1');
+    });
+
+    it('should throw if we add localization key to a component that already has it', async () => {
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+      const tree = await runner.runSchematic('add-localization-key', {
+        path: o3rComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, initialTree);
+      await expect(runner.runSchematic('add-localization-key', {
+        path: o3rComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, tree)).rejects.toThrow();
+    });
+
+    it('should throw if we add localization key to a component that is not localized', async () => {
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+      await expect(runner.runSchematic('add-localization-key', {
+        path: emptyO3rComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, initialTree)).rejects.toThrow();
+    });
+  });
+
+  describe('Angular component', () => {
+    beforeEach(() => {
+      initialTree = Tree.empty();
+      initialTree.create(ngComponentPath, `
+        import {CommonModule} from '@angular/common';
+        import {Component} from '@angular/core';
+
+        @Component({
+          selector: 'ng-test',
+          standalone: true,
+          imports: [CommonModule],
+          template: ''
+        })
+        export class NgComponent {}
+      `);
+      initialTree.create('.eslintrc.json', fs.readFileSync(path.resolve(__dirname, '..', '..', 'testing', 'mocks', '__dot__eslintrc.mocks.json')));
+    });
+
+    it('should throw if no Otter component', async () => {
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+
+      await expect(runner.runSchematic('add-localization-key', {
+        path: ngComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, initialTree)).rejects.toThrow();
+
+      await expect(runner.runSchematic('add-localization-key', {
```

Comment:
I thin this expect should be in a dedicated test

Reply:
as discussed will be done in a separate pr on all schematics